File: java/src/main/java/heaps/FindKthLargest.java
package heaps;

import java.util.PriorityQueue;

public class FindKthLargest {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> maxHeap = new PriorityQueue<>((a, b) -> b - a);

        for (int num : nums) {
            maxHeap.add(num);
        }

        int res = 0;
        while (k > 0) {
            res = maxHeap.poll();
            k--;
        }

        return res;
    }

    public static void main(String[] args) {

    }
}

File: python/heaps/findKthLargest.py
from typing import List
import heapq

def findKthLargest(nums: List[int], k: int) -> int:
    maxHeap = []

    for num in nums:
        maxHeap.append(-num)
    
    heapq.heapify(maxHeap)
    
    num = 0
    while(k > 0):
        num = heapq.heappop(maxHeap)
        k -= 1

    return -num
    
    



nums = [-1,-1]
k = 2
findKthLargest(nums, k)
